Tools/Scripts/compare-static-analysis-results

```diff
@@ -160,31 +166,48 @@ def compare_project_results_by_run(args, archive_path, new_path, project):
     new_files_total = set()
     fixed_issues_total = set()
     fixed_files_total = set()
+    project_results_passes = {}
+    project_results_failures = {}
 
     for checker in CHECKERS:
-        print(f'{checker}:')
         new_issues, fixed_issues = find_diff(args, f'{archive_path}/{checker}Issues', f'{new_path}/{project}/{checker}Issues')
         new_files, fixed_files = find_diff(args, f'{archive_path}/{checker}Files', f'{new_path}/{project}/{checker}Files')
         fixed_issues_total.update(fixed_issues)
         fixed_files_total.update(fixed_files)
         new_issues_total.update(new_issues)
         new_files_total.update(new_files)
 
-        print(f'    Issues fixed: {len(fixed_issues)}')
-        print(f'    Files fixed: {len(fixed_files)}')
-        print(f'    New issues: {len(new_issues)}')
-        print(f'    New files with issues: {len(new_files)}\n')
+        project_results_passes[checker] = list(fixed_files)
+        project_results_failures[checker] = list(new_files)
+
+        unexpected = False
+        results_summary = ''
+        for type, type_list in {
+            'Issues fixed': fixed_issues,
+            'Files fixed': fixed_files,
+            'Unexpected issues': new_issues,
+            'Unexpected failing files:': new_files
+        }.items():
```

Comment:
Hm... I'm not sure that this format is more readable.
A bunch of print statements looked simpler.

Reply:
Oh I meant for the logging output. Before we had lines for each checker even though there were no unexpected results. Now they only print if there are results.
I agree with the additional complexity, but now we can iterate and check each list.

Reply:
Sure, I guess we can list all the counts as long as one is nonzero.

Reply:
oh I see. We're only putting a non-zero line. I guess we can keep this code as is in that case.

Reply:
Lol I just made your changes, I think my implementation really only saves one line or two and it helps to have standardized output for each checker so you don't need to read a line to figure out which one it is (if people will be looking at this at all).